core/base-service/route.js

```diff
@@ -67,9 +69,19 @@ function namedParamsForMatch(captureNames = [], match, ServiceClass) {
   return result
 }
 
+function getQueryParamNames({ queryParams = [], queryParamSchema }) {
+  if (queryParamSchema) {
+    const { children, renames = [] } = Joi.describe(queryParamSchema)
```

Comment:
I didn't know `Joi.describe` was a thing. Just another chapter in my "today the thing I learned about Joi was..." book